lib/gollum-lib/filter/code.rb

```diff
@@ -126,14 +136,34 @@ def remove_leading_space(code, regex)
       code.gsub!(regex) { '' }
     end
   end
+  
+  def generate_placeholder_id(extra, code)
+    id = "#{open_pattern}#{Digest::SHA1.hexdigest("#{extra}.#{code}")}#{close_pattern}"
+  end
+  
+  def cache_rblock(extra, code, indent = '')
+    id = generate_placeholder_id(extra, code)
+    @map[id] = {:output => "#{indent}```#{extra}\n#{code}\n```"}
+    id
+  end
 
-  def cache_codeblock(language, code, indent = "")
+  def cache_codeblock(language, code, indent = '')
     language = language.to_s.empty? ? nil : language
-    id = "#{open_pattern}#{Digest::SHA1.hexdigest("#{language}.#{code}")}#{close_pattern}"
+    id = generate_placeholder_id(language, code)
     cached = @markup.check_cache(:code, id)
     @map[id] = cached ?
       { :output => cached } :
       { :lang => language, :code => code, :indent => indent }
     id
   end
+  
+  def render_r(content)
```

Comment:
Can you explain how the resulting figure is dealt with? Is the image stored somewhere on disk? Or is it `base64` encoded and placed directly in the html fragment?

Reply:
The latter! `knitr` ensures every fragment is rendered directly "on screen".

Reply:
That's very neat! I must have missed that option when researching this before.